json prepared for ingest api now allows arbitrary key value pairs

sanzog03 opened the pull request:
**Summary:** Summary of changes

Addresses [sub-feature of GHGC-269: Fix workflows api for GHGC](https://github.com/US-GHG-Center/ghgc-architecture/issues/269)

## Changes

* Allow arbitrary key value pairs in the request body payload

sanzog03 committed:
json prepared for ingest api now allows arbitrary key value pairs

slesaad commented on workflows_api/runtime/src/schemas.py:
```diff
@@ -181,6 +182,9 @@ def object_is_accessible(cls, values):
         )
         return values
 
+    class Config:
+        extra = Extra.allow
+
```
this shouldn't need extras

sanzog03 replied:
This one is for `S3Input` and if we need to allow extra attributes inside discovery_items, we need this as well.

slesaad replied:
we don't need to allow extra attributes inside `discovery_items`

slesaad commented on workflows_api/runtime/src/schemas.py:
```diff
@@ -262,6 +268,8 @@ def check_sample_files(cls, values):
             )
         return values
 
+    class Config:
+        extra = Extra.allow
```
does this not get inherited from `Dataset` ?

sanzog03 replied:
makes sense. removing it.

sanzog03 replied:
makes sense. Removing it!

slesaad commented on workflows_api/runtime/src/schemas.py:
```diff
@@ -280,3 +288,6 @@ def only_one_discover_item(cls, discovery_items):
             )
         return discovery_items
+
+    class Config:
+        extra = Extra.allow
```
does this not get inherited from `Dataset` ?

sanzog03 replied:
makes sense. removing it.

sanzog03 replied:
makes sense. Removing it!

slesaad commented on workflows_api/runtime/src/collection_publisher.py:
```diff
@@ -58,21 +58,25 @@ class Publisher:
         self.func_map = {
-            DataType.zarr: self.create_zarr_collection,
-            DataType.cog: self.create_cog_collection,
+            DataType.zarr.value: self.create_zarr_collection,
+            DataType.cog.value: self.create_cog_collection,
```
🤔

sanzog03 replied:
the expected map is between string key and function value. It was rather <Datatype.cog> as key and function as value.

sanzog03 commented on workflows_api/runtime/src/collection_publisher.py:
```diff
@@ -58,21 +58,25 @@ class Publisher:
+        })
+        # Convert enum values to their string equivalents
+        for key, value in dataset_dict.items():
+            if isinstance(value, DataType):
+                dataset_dict[key] = value.value
```
there was a key value namely `datatype: <Datatype.cog>`
It was not a valid json for the ingest api call

sanzog03 replied:
The value for the key `data_type` in the collection is an enumerator. Its value will be `<DataType.cog>` by default, which is not a valid JSON and will cause an issue during API call for ingest. Previously, this was ignored in the JSON body for ingest, However, as we will allow everything now, this will come in as well. So fixed it here.

sanzog03 reviewed (commented):
replies to the previous comments

sanzog03 committed:
removed the ones inherited from parent

sanzog03 committed:
1. added global value exception handler
2. filter out unnecessary attributes that doesnot align with the stac specs
3. user provided attribute has greater precedence than the created one
